src/main/java/greencity/entities/Category.java

```diff
@@ -0,0 +1,36 @@
+package greencity.entities;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Category {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true, length = 100)
+    private String name;
+
+    @ManyToOne
+    private Category category;
```

Comment:
Category class has a category field?
Why?

Reply:
This is named Recursion Reference. One category can has many sub-category: Food-category has cafe-category and more. And cafe-category can has many sub-categories.

Reply:
First of all - you don't need categories to have such a tree-like structure. In business logic of your app there is only need for a main category (such as come-with-own-cup, e-charge, etc) and some of these categories can have (or cannot) sub-categories (refering to the #21 issue come-with-own-cup can have type(sub-category) such as cafe, restaurant, coffee-shop etc) and that is it. You never should go deeper in this hierarchy of types in your app, so Recursion reference is an overkill for your app and it will be more complicated to work with in the future(read acceptance criteria for #21 issue that you need to choose type from dropdown menu)

So you can make two tables Category and Sub-Category with OneToMany relation.

OR if you want to save the possibility to have unlimited depth tree structure, at least name this field parentCategory, as you did in the Comment entity
But again, in Comment Recursion reference is obvious to use and mathes business logic. Here it is an overkill